cmd/snapd-apparmor/main.go

```diff
@@ -68,19 +69,27 @@ import (
 // any loss of functionality. This is an unsupported configuration that cannot
 // be properly handled by this function.
 func isContainerWithInternalPolicy() bool {
-	if release.OnWSL {
-		// WSL-1 is an emulated Windows layer that has no support for AppArmor.
-		// WSL-2 is a virtualised environment with the Linux kernel as
-		// distributed by Microsoft.  In the future, if Microsoft enables
-		// AppArmor in their kernel configuration and adjusts their special
-		// init process, which launches individual distributions in
-		// quasi-containers, to initialize apparmor nesting, we might re-enable
-		// this and treat WSL-2 like a LXD/Incus container, with nested
-		// security.
-		return false
+	var securityFSPath = filepath.Join(dirs.GlobalRootDir, "/sys/kernel/security")
+
+	if release.OnWSL && release.WSLVersion == 2 && osutil.IsDirectory(securityFSPath) {
+		// WSL-1 is an emulated Windows layer that has no support for
+		// AppArmor.  WSL-2 is a virtualised environment with the Linux
+		// kernel as distributed by Microsoft.
+		//
+		// In the future, Microsoft might enable AppArmor in their
+		// kernel configuration and might adjust their special init
+		// process, which launches individual distributions in
+		// quasi-containers, to initialize apparmor nesting and this
+		// logic may go away entirely (the special case for WSL).
+		//
+		// In the meantime, given that people experiment with appamor
+		// on WSL so we only bail out if the securityfs is not mounted
+		// and assume everything else is "just right" even though that
+		// is not really true.
+		return true
```

Comment:
Should this also keep the `return false` behaviour previously for other WSL cases? ie:

````
if release.OnWSL {
  if release.WSLVersion == 2 && osutil.IsDirectory(securityFSPath) {
    return true
  }
  return false
}
```

Reply:
I will kind of fail by itself on one of the checks below but I can make it explicit to make the change less risky.

Reply:
Done